state-chain/chains/src/sol.rs

```diff
@@ -0,0 +1,142 @@
+pub use cf_primitives::chains::Solana;
+use cf_primitives::{AssetAmount, ChannelId};
+
+use sp_core::ConstBool;
+use sp_std::vec;
+
+use sol_prim::SlotNumber;
+
+use crate::{address, assets, FeeRefundCalculator, TypeInfo};
+use codec::{Decode, Encode, MaxEncodedLen};
+use serde::{Deserialize, Serialize};
+
+use super::{Chain, ChainCrypto};
+
+pub mod api;
+pub mod benchmarking;
+pub mod consts;
+mod tracked_data;
+
+pub use sol_prim::{
+	pda::{Pda as DerivedAddressBuilder, PdaError as AddressDerivationError},
+	Address as SolAddress, Digest as SolHash, Signature as SolSignature,
+};
+
+pub use tracked_data::SolTrackedData;
+
+impl Chain for Solana {
+	const NAME: &'static str = "Solana";
+	const GAS_ASSET: Self::ChainAsset = assets::sol::Asset::Sol;
+
+	type ChainCrypto = SolanaCrypto;
+	type ChainBlockNumber = SlotNumber;
+	type ChainAmount = AssetAmount;
+	type TransactionFee = Self::ChainAmount;
+	type TrackedData = tracked_data::SolTrackedData;
+	type ChainAsset = assets::sol::Asset;
+	type ChainAccount = SolAddress;
+	type EpochStartData = ();
+	type DepositFetchId = ChannelId;
+	type DepositChannelState = ();
+	type DepositDetails = ();
```

Comment:
It is not immediately obvious which ones we will need. I will add a todo comment in front each of the empty types so that we can revisit them once all of the sol code is plugged in.